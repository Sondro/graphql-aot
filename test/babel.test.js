const babel = require('@babel/core')

test('main', () => {
  const { code } = babel.transform(`
  graphql\`
    allFiles {
      id
    }
  \`
  `, {
    babelrc: false,
    plugins: [
      require.resolve('../lib/babel')
    ]
  })

  expect(code).toContain('graphql-aot/lib/noop.gql?query=%0A%20%20%20%20allFiles%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%7D%0A%20%20')
})

describe('importFrom', () => {
  it('no import', () => {
    const { code } = babel.transform(`
  graphql\`
    allFiles {
      id
    }
  \`
  `, {
    babelrc: false,
    plugins: [
      [require.resolve('../lib/babel'), {
        importFrom: 'foo'
      }]
    ]
  })

    expect(code).toMatchSnapshot()
  })

  it('has import', () => {
    const { code } = babel.transform(`
  import { graphql } from 'foo'

  graphql\`
    allFiles {
      id
    }
  \`
  `, {
    babelrc: false,
    plugins: [
      [require.resolve('../lib/babel'), {
        importFrom: 'foo'
      }]
    ]
  })

    expect(code).toMatchSnapshot()
  })
})
