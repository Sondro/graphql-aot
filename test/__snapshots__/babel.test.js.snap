// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importFrom has import 1`] = `"require(\\"/Users/egoist/dev/graphql-aot/lib/noop.gql?query=%0A%20%20%20%20allFiles%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%7D%0A%20%20\\");"`;

exports[`importFrom no import 1`] = `
"graphql\`
    allFiles {
      id
    }
  \`;"
`;
